src/client/parser.rs

```diff
@@ -572,8 +572,10 @@ impl<M, S> Protocol for Parser<M, S>
                         idle: true,
                     }, scope), scope)
             },
-            ReadHeaders { machine, request, .. } => {
-                let mut req = request.with(transport.output());
+            ReadHeaders { machine, request, is_head } => {
+                let mut req: Request = request.with(transport.output());
+                req.1 = is_head;
```

Comment:
@tailhook If I don't include this line then machines in the `ReadHeaders` state will panic here: https://github.com/tailhook/rotor-http/blob/master/src/client/parser.rs#L348

How should I test this?

Sorry for making such a fuss in this PR, `rotor` is still a bit of a mystery to me 😄